### PR TITLE
LOG-8039 update packages

### PR DESCRIPTION
**Pull request by alestrunda**
update packages, readme, code format

### CODE REVIEW

File: lib/winston-loggly.js [hunk 1]
```diff
@@ -95,34 +97,31 @@ winston.transports.flushLogsAndExit = flushLogsAndExit;
 //
 Loggly.prototype.name = 'loggly';
 
-const validateMetadata = (meta) => {
+const validateMetadata = meta => {
   if (meta == null) {
     return {};
   } else if (typeof meta !== 'object') {
-    return { metadata: meta };
+    return { Metadata: meta };
```
Comment (michal-bures) [inline lib/winston-loggly.js]: Woah, why the change to upper case? That would be breaking change for many users right?

File: lib/winston-loggly.js [hunk 2]
```diff
@@ -134,15 +133,16 @@ Loggly.prototype.log = function (meta, callback) {
...
+    callback && callback(err, result);
   }
 
-  result = (meta && meta.tags)
-    ? this.client.log(message, meta.tags, logged)
-    : this.client.log(message, logged);
-    
-  callback();
+  result =
+    meta && meta.tagsZ
```
Comment (michal-bures) [inline lib/winston-loggly.js]: `tagsZ`? typo?

File: README.md [hunk 3]
```diff
@@ -82,54 +77,40 @@ Here is an example of how to use the method:
...
+All of the winston tests are written in [vows][5], and cover all of the use cases described above. You will need to add valid credentials for the various transports included to `test/config.json` before running tests:
 
 ``` js
   {
     "transports": {
       "loggly": {
         "subdomain": "your-subdomain",
-        "inputToken": "really-long-token-you-got-from-loggly",
+        "token": "really-long-token-you-got-from-loggly",
         "auth": {
```
Comment (michal-bures) [inline README.md]: This part appears to be pointless? It's not used by the tests. Maybe someone planned to login to loggly API and verify the logs arrived as part of the tests... but there's no such thing implemented. Let's remove it from here and from the sample config.

File: package.json [hunk 4]
```diff
@@ -27,19 +28,22 @@
...
   ],
   "dependencies": {
-    "node-loggly-bulk": "^2.0.1",
-    "clone": "^2.1.1",
-    "winston": "^3.0",
-    "winston-transport": "^4.2.0"
+    "node-loggly-bulk": "^2.2.4",
+    "clone": "^2.1.2",
+    "winston": "^3.2",
+    "winston-transport": "^4.3.0"
```
Comment (michal-bures) [inline package.json:34]: These two should really really be in peer dependencies, we shouldn't force users to use our chosen version of winston... But moving it there can break stuff for existing users which will now have to add the winston dependency manually... Let's make a separate task for that.

**Review by michal-bures — CHANGES_REQUESTED** [review]
Noticed a few issues, but otherwise seems ok, was able to run the tests and see results in loggly.

Don't forget to remove the `alpha` version from package.json before merge (and best not to merge before we verify the functionality in portal).

The automated tests don't give me too much confidence (as illustrated by the two issues I found in code which didn't break any of them), but expanding them is a topic for another task.

File: README.md [hunk 5]
```diff
@@ -25,13 +22,11 @@ Please note that the documentation below is for `winston-loggly-bulk@2.x`. [Read
...
-The Loggly transport is based on [Nodejitsu's][2] [node-loggly][3] implementation of the [Loggly][0] API. If you haven't heard of Loggly before, you should probably read their [value proposition][4]. The Loggly transport takes the following options. Either 'inputToken' or 'inputName' is required:
+The Loggly transport is based on [Nodejitsu's][2] [node-loggly][3] implementation of the [Loggly][0] API. If you haven't heard of Loggly before, you should probably read their [value proposition][4]. The Loggly transport takes the following options.
 
-* __level:__ Level of messages that this transport should log.
 * __subdomain:__ The subdomain of your Loggly account. *[required]*
-* __auth__: The authentication information for your Loggly account. *[required with inputName]*
-* __inputName:__ The name of the input this instance should log to.
-* __inputToken:__ The input token of the input this instance should log to.
+* __token:__ The access token (previously inputToken) *[required]*
+* __auth__: The authentication information for your Loggly account.
```
Comment (michal-bures) [inline README.md:29]: This is normally not needed right? Actually is basic auth supported at all anymore? I recommend testing it and if it doesn't work remove this.

Especially since "token" is *required*, there seems to be no point in entering username/password

Comment (alestrunda) [reply README.md:29]: Again the package uses node-loggly-bulk and there are both token and auth. Token is required, auth seems to be useless, but still I wasn't sure whether it's save to remove it completely.

File: package.json (hunk 4 above)
Comment (alestrunda) [reply package.json:34]: Agree.

File: README.md (hunk 3 above)
Comment (alestrunda) [reply README.md]: Yes, pointless. Anyway, the package uses node-loggly-bulk and there is auth used, so I decided not to remove it. If it's useless node-loggly-bulk should be reviewed as well.

File: lib/winston-loggly.js (hunk 2 above)
Comment (alestrunda) [reply lib/winston-loggly.js]: typo indeed

File: lib/winston-loggly.js (hunk 1 above)
Comment (alestrunda) [reply lib/winston-loggly.js]: I was doing some tests because of the "metadata" prop issue I reported, forgot to put it back, it should be lowercase of course.

**Review by michal-bures — APPROVED** [review]
Looks good, don't forget to remove `alpha2` from `package.json` before merging 🙂.
